velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -319,11 +428,24 @@ void PageReader::prepareDataPageV1(const PageHeader& pageHeader, int64_t row) {
 
     return;
   }
-  pageData_ = readBytes(pageHeader.compressed_page_size, pageBuffer_);
-  pageData_ = decompressData(
-      pageData_,
-      pageHeader.compressed_page_size,
-      pageHeader.uncompressed_page_size);
+  if (data_qpl_job_id != 0 && pre_decompress_data && job_success) {
```

Comment:
It's too invasive, in pagereader we shouldn't have any logic related to specific codec. Later we may have QAT related code, then some other accelerators. It will make the normal s/w path too complex. Can we abstract the PageReader to move the logic to something like IAAPageReader?

Reply:
@FelixYBW  According the the previous comments by [Yuhta](https://github.com/Yuhta) [on Aug 21](https://github.com/facebookincubator/velox/pull/6176#discussion_r1300170969)
The duplicate here with PageReader is just too large. All you need on interface level is a preDecompRowGroup. Please use leverage inheritance or templating to reduce the duplication.

Reply:
you shouldn't duplicate the file, add the abstract layer, reuse the existing code as much as possible

Reply:
If we want to reuse the code, there will be much code such as PageReader will change too.